org.jacoco.core/src/org/jacoco/core/internal/flow/Instruction.java

```diff
@@ -65,20 +69,35 @@ public void addBranch() {
 	 * @see #addBranch()
 	 * @param predecessor
 	 *            predecessor instruction
+	 * @param branch
+	 *            branch number in predecessor that should be marked as covered
+	 *            when this instruction marked as covered
 	 */
-	public void setPredecessor(final Instruction predecessor) {
+	public void setPredecessor(final Instruction predecessor,
```

Comment:
@Godin Thanks, hopefully helps us and others with this tricky piece of code in future.